### DISCUSSION TITLE
Suggestion: Change behaviour of the «expand signal menu» btn

### DISCUSSION THREAD

**Discussion by Rau117** [OpenTTD/OpenTTD#9687] [Features]
I didn't notice this the first time, but now I can clearly see that new signals button is completely useless. This is certainly better than looking among the advanced settings, but will be pressed not more than 2-3 times and will be forever forgotten.

If the player knows about the existence of block signals and want to use it, he will permanently turn on the old (extended) menu and completely forget about this button.

If the player does not know about block signals, he will not turn on the old menu and will also forget about this button. Or if he is inquisitive, he will press this button and will… "use block signals incorrectly."

I suggest to show a message asking the player to choose how the signal menu will look and act.
Something like 

> Do you want to show advanced signals menu? 
> Right click on answer to see description. 
> You can change this in advanced settings
> Yes     No      Not now

If the player clicks YES, show the old signals menu, which contains all the signals.
By the new button (which was added in 12.1) one row of the signals will be hided. Click on «Signal convert» button to change the signals row. Ctrl+click will hold this button and it will acts as normal «Signal convert» funcion (more on that later).
![изображение](https://user-images.githubusercontent.com/47086324/141056010-ec0019b1-0894-4a89-8bc6-752d5deecc49.png)

If the player clicks NO, the signal menu will be turned off completely. The semaphore / traffic light is selected according to the setting «build semaphores before». The default signal is one-way PBS, changing the signal by ctrl+click on curren signal. (as it was before 12.0 with several settings on it).
Ctrl+click on the signall menu on the rail toolbar will open the simplified signal menu in its current (12.1) form, where you can change signal to build and distance between signals. Make distance between signals = 5 by default for the NEW players.
![изображение](https://user-images.githubusercontent.com/47086324/141056521-681aab39-0a14-4f09-b4de-f69869d1833f.png)

About «Signal convert».
It is not clear why this function is needed. It works extremely poorly and unclear.
It is can be used to change a single signal in the few places, but it is absolutely unacceptable for a massive usage. 
It is not necessary to remove it at all, it helps those who are too lazy to reach for the ctrl button or ... who do not have it (this is a slightly different question, virtual modifier buttons). But definitely worth adding the ability to change the type of signals to the opposite (sem←→light) on the entire connected path/area/similar to «convert rail» tool.

A short video in which I trying to use this feature: https://youtu.be/LD3UGMkQ32g 

It is also worth applying the mechanics of placing PBS on ordinary block signals. 
So that a one-way block signal is set with a single click, signal changes its direction on the second click, and on the third click the signal becomes two-sided. For the simplest block signal ONLY. Leave the pre-signal arrangement as it is. 
And of course add a setting to this. Nobody likes permanent changes that affect habits and «muscle memory».

**Comment by LordAro** [comment]
My response in your original issue was overly terse, I apologise for that.

There's several suggestions/issues here that I can see.

* Show advanced signals button
I'm confused about what you want. The button was added in response to comments (mostly from yourself) about how the setting to reenable signals was hard to find. But now you don't want a way to enable them? A pop up window isn't viable - we can't possibly add such a mechanism just for signals, nor do we have any feasible way of detecting when the game has been upgraded
(Improving the settings window is a separate, ongoing, discussion)

* Convert signals button
Might be some legitimate issues here. I agree that its use is limited - I don't think I've ever used it

* Hide semaphores
This has come up several times, we've never been able to come up with a good way of switching between the two. Perhaps the convert signal button could be converted to toggle between the two types, and overbuilding would "just work" (tm)? That might work.

* Change default signal separation
Why? You haven't provided any justification here

* Signal window looks strange with BRIX
This is the actual size of the sprites. They appear to have a lot of blank space (I'd consider it an issue of the NewGRF) - there's no way to detect the transparency without a lot of effort (it would be slow).

* Place one-sided block/pre-signals first
This seems like a good suggestion to me

And as ever:
[![](https://imgs.xkcd.com/comics/workflow.png)](https://xkcd.com/1172/)
We *cannot* add settings for every usecase. We have too many already and are actively working on reducing them. Sorry about that.

**Reply by Rau117** [reply]
Why reduce the number of settings? To reduce the number of play styles? Or to reduce the convenience of the game? 
Or just for newbies, who ask too many questions about settings?

Let's then just hide these settings from "public" access. Make «super-advanced» settings visible only after changing the hidden (from the normal gameplay) parameter in the openttd.cfg file.
Normal players do not change configuration file, normal players just use normal settings. 
However, in the game settings there are already categories that hide «weird» settings. Why not leave it as it is? Who exactly is hindered by a large selection of settings?

One more thing. Why the «signal by default» setting was removed? It's quite annoying to change signal to block after each time I reload my game (or reconnect to the server). Is this the setting that needs to be removed?
And I am not alone in this. This hinders many players on our servers (in-game/discord). But not so many know that they can ask the developers about it.

**Reply by LordAro** [reply]
>Who exactly is hindered by a large selection of settings?

We are. The hundreds of settings interact in extremely hard to predict ways, and testing all of the millions of possibilities is literally impossible. Reducing the number of rarely changed settings makes the game easier to maintain, and has the added bonus of making the game easier to understand to the average person.

**Reply by same-f** [reply]
Well, let's leave only two on and off buttons in the game. 

Then you will have more time to rest and players will have no questions about the settings

**Comment by ldpl** [comment]
IMO instead of asking the player ridiculous questions, it would be much better to show some kind of tutorial thing like in #9474.

**Comment by LC-Zorg** [comment]
**Show advanced signals button**
I don't think this is helpful for new players. Asking them questions in a field they don't know yet won't make anything easier for them. And when it comes to showing tips, a great many, even much simpler games, have a tutorial mode. Meanwhile, here in much more complex OTTD isn't and it would be very advisable to create a game mode where the player could learn step by step all, or at least the most important elements of the game, such as "Building roads / tracks"> "Putting basic signaling"> "Creating basic orders" > "Vehicle duplication"> "Advanced signaling"> etc. There, when performing tasks, such larger messages could appear, even with graphics. But... well, realistically, it's more of a fantasy zone, but the **Help and manuals window project** #7786 is open and that could also be of great help to new players.
Another idea might be to add (exclusive) question mark icons where the player might have some problems or doubts - even anywhere. I showed something in this direction on one of the versions of the improved finance window - [here](https://user-images.githubusercontent.com/67662046/119268275-8010da80-bbf2-11eb-95c4-daa14e7d2c71.png)

**Convert signals button**
As it stands, it's one of the most useless features in the game for me. It doesn't matter much to me, but I agree with Rau here that this button would be much more useful if the signal shift was area-wise and not one-piece. Then it would be useful to be able to disable the selection of a specific signaling device (or by default any wouldn't be selected), thanks to which it would be possible to simultaneously convert different types of semaphores to signaling devices and vice versa.
![Semaphores to Signals convert](https://user-images.githubusercontent.com/67662046/141849144-7a23d3cd-eddc-4cb5-9b1d-cd4945232f1c.gif)

**Hide the semaphores**
You don't like it, but for me the best way would be to add an option to disable semaphores entirely. Many players don't want to use them at all. Alternatively! What would be simpler, if somewhat hidden, might also be that setting the signals appearance date to year 0 will disable the semaphores (also using Ctrl while building will not build the semaphore). Sample explanation for setting: "Choose 0 to disable semaphores completely."

**First, place a one-sided block/pre-signal**
Yes, I think it could be better than it is now. Using the bidirectional variant usually only ends in problems and you have to know how to use it, so it's not good that it's the default. I also think that this behavior should apply to all signals. Certainly a bidirectional entry signal doesn't make sense as a basic option. Then only two out of five remain, where the use of unidirectional is similarly frequent or even more frequent than the use of bidirectional. From my point of view, this is not a big change. While it is about player habits, it seems beneficial enough that it will be fairly easy for players to get used to, so I don't think it's necessary to add options.

**"We have too many already (settings) and are actively working to reducing them."**
I understand the reason, but... heh... I think that's the wrong approach. Perhaps there has been more of this recently, but I've known two instances of deleting a setting, and both are really frustrating.

The first case is to remove the option of no planes crashes. First, it destroyed a lot of my save files, forcing me to use the cheat option, which I really don't like and don't think I'm alone in that. Second, I am currently working on updating two scenarios for two popular online servers and this change has prevented me from using the best setting whereby the profits from transporting passengers by air would be reasonably limited while transporting other goods could still generate some profit. With this option removed, there is no way to reasonably balance it - there is always some shit. It's really discouraging.

The second case was the last deletion of the default signaling type setting. Generally, it doesn't matter to me, because I use all of them anyway and often start with something other than the default one. However, the frustrating thing is that now each time I run a different version of the game, the signaling settings are different - the ones that limit the availability of signaling types are constantly changing. I have to go into these settings and tweak them every time. I was a supporter of simplifying the signaling for beginners, but the way to make this change is really terrible.

In the past, other settings were probably also removed, which unfortunately means that many scenarios and saves are unplayable today.

**Reply by Rau117** [reply]
> Hide the semaphores
> You don't like it, but for me the best way would be to add an option to disable semaphores entirely. 

Semaphores do not always perform their intended purpose – being old-style signals for early game. In some NewGRFs, at least in PURR and NUTS semaphores can be replaced with modern signals of a different appearance.
![изображение](https://user-images.githubusercontent.com/47086324/142027883-835fd2ce-df45-4228-b8a2-5f39ae5f8071.png)

Completely removing semaphores (from the menu, of course) will not be as effective as making a button to change the appearance in the signals menu.
Or we can go further and make it possible to add several types of signals through NewGRF, and allow you to select apperance of signals from the list.
But this is a different question and it need to think about how it all will look… if someone ever agrees to create such an opportunity 

Also, due to the peculiarities of using pre-signals, their placement should be left as is - the first click puts 2 signals, the second and third - one-way variants. For now they are used as logical elements or priority; for their intended purpose, they are easily replaced by PBS.

For ctrl+click on an empty rail I suggest... oh, you won't like it... I suggest adding a setting that allows you to choose which signal will be placed. _Ctrl + click on an existing signal should act exactly as it does now._

For example, I want to use block signals everywhere, but set a **two**-way path signal at stations - I select a block signal in the signals menu, and choose a two-way route signal in the settings, the convenience and speed of building stations will be increased several times. Right now I have to select the default two-way route and change signals through CTRL + a few clicks.
_I don’t know how others are building, but for me personally, this strategy is the most convenient and fast._

A small demonstration of the number of clicks per one station. 
_This station is the most efficient + easiest to build for casual play._
![2021-11-16 20-12-33](https://user-images.githubusercontent.com/47086324/142036942-20de43d3-295f-4dfc-a277-a8244d9545b5.gif)
Unfortunately, there is no signal placement sound in this gif.

**Reply by LC-Zorg** [reply]
>Completely removing semaphores (from the menu, of course) will not be as effective as making a button to change the appearance in the signals menu.

So it can be do this way...
1. Add two options to the "Show signal types" setting, which would give a total of four:
- Path signal only
- All signals
- (+) Path signal only, but semaphores
- (+) All signals, but semaphores
2. Then make the toggle show all or only those that the player has selected in the setting above.

>Or we can go further and make it possible to add several types of signals through NewGRF, and allow you to select apperance of signals from the list.

I think it is so niche that it is not worth it.

Coming back to the deletion of settings, I think "Cycle through signal types" is a very good candidate. I don't know what the point is to have a limited choice over what is visible. For me it is only confusing.

**Reply by Rau0x75** [reply]
Doesn't sound very convenient. If you hide the semaphores in the settings, the anger of the players will be similar to «hide advanced™ signals» in 12.0. 

In "realistic" or "eye candy" games, semaphores are used quite actively. Actively enough to make it annoying to constantly change setting.

But if you change the behavior of the button (the one that you showed on the gif), it will be much better. Something like simple click on the button turns on "change signals",  ctrl + click changes the signals panel.

_I was finally banned ..._
